domain/controllerconfig/state/state.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/domain"
+)
+
+// State represents a type for interacting with the underlying state.
+type State struct {
+	*domain.StateBase
+}
+
+// NewState returns a new State for interacting with the underlying state.
+func NewState(factory domain.DBFactory) *State {
+	return &State{
+		StateBase: domain.NewStateBase(factory),
+	}
+}
+
+// Add ....
+func (s *State) Add(ctx context.Context, key string, value string) error {
```

Comment:
Maybe we could find a more homogeneous naming for these methods in `State`, since they will be repeated all over the `domain/*/state` and probably most will be inside the CRUD set of operations. For instance, in `domain/modelmanager` it's called `Create()`.